HPC/WRF.md

```diff
@@ -0,0 +1,263 @@
+export LD_LIBRARY_PATH=${INSTALLDIR}/gcc-${GCC_VERSION}/lib64:$LD_LIBRARY_PATH
+export PATH=${INSTALLDIR}/openmpi-${OPENMPI_VERSION}/bin:$PATH
+export LD_LIBRARY_PATH=${INSTALLDIR}/openmpi-${OPENMPI_VERSION}/lib:$LD_LIBRARY_PATH
+export CC=mpicc
+export CXX=mpicxx
+export FC=mpif90
+export F77=mpif90
+export F90=mpif90
+export HDF5=${INSTALLDIR}/hdf5
+export PHDF5=${INSTALLDIR}/hdf5
+export NETCDF=${INSTALLDIR}/netcdf
+export PNETCDF=${INSTALLDIR}/pnetcdf
+export PATH=${INSTALLDIR}/netcdf/bin:${PATH}
+export PATH=${INSTALLDIR}/pnetcdf/bin:${PATH}
+export PATH=${INSTALLDIR}/hdf5/bin:${PATH}
+export LD_LIBRARY_PATH=${INSTALLDIR}/netcdf/lib:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${INSTALLDIR}/pnetcdf/lib:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${INSTALLDIR}/hdf5/lib:$LD_LIBRARY_PATH
+export WRFIO_NCD_LARGE_FILE_SUPPORT=1
+export NETCDF_classic=1
+
+# configure WRF, choose option 8 '(dm+sm) GCC (gfortran/gcc): Aarch64' and then option 1 for 'basic' nesting
+./configure
+
+# compile WRF
+./compile -j $(nproc) em_real 2>&1 | tee compile_wrf.out
+```
+  * download CONUS 2.5km and 12km dataset
+```
+wget https://www2.mmm.ucar.edu/wrf/src/conus12km.tar.gz
+wget https://www2.mmm.ucar.edu/wrf/src/conus2.5km.tar.gz
+tar xf conus12km.tar.gz
+tar xf conus2.5km.tar.gz
+
+# also copy the following files from $wrf_install/run directory to conus12km/conus2.5km directory.
+# *.TBL, *.formatted, RRTMG* and CAMtr_volume_mixing_ratio
+```
+  * run WRF simulation with slurm scheduler. We found that using 8 threads per rank and setting threads affinity to be "compact" gives the best performance. 
+```
+#!/bin/bash
+#SBATCH --wait-all-nodes=1
+#SBATCH --ntasks-per-node=8
+#SBATCH --cpus-per-task=1
+#SBATCH --nodes=4
+#SBATCH --ntasks-per-core=1
+#SBATCH --export=ALL
+#SBATCH --partition=compute
+#SBATCH --exclusive
+#SBATCH -o /shared/data/conus12km-uncompressed/slurm.out
+
+#ENV VARIABLES#
+
+#---------------------Run-time env-----------------------------------------
+ulimit -s unlimited
+
+export OMP_STACKSIZE=12G
+export OMP_NUM_THREADS=8
+export KMP_AFFINITY=compact,verbose
+
+wrf_install=/shared/wrf-arm-efa
+export PATH=${wrf_install}/gcc-10.2.0/bin:$PATH
+export LD_LIBRARY_PATH=${wrf_install}/gcc-10.2.0/lib64:$LD_LIBRARY_PATH
+
+export PATH=${wrf_install}/openmpi-4.1.4/bin:$PATH
+export LD_LIBRARY_PATH=${wrf_install}/openmpi-4.1.4/lib:$LD_LIBRARY_PATH
+
+export LD_LIBRARY_PATH=${wrf_install}/netcdf/lib:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${wrf_install}/pnetcdf/lib:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${wrf_install}/hdf5/lib:$LD_LIBRARY_PATH
+
+#--------------------------------------------------------------------------
+echo "Running WRF on $(date)"
+cd /shared/data/conus2.5km
+echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope
+
+date -u +%Y-%m-%d_%H:%M:%S >> wrf.times
+mpirun --map-by socket:PE=8 --bind-to core ./wrf-v4.5.exe &>> wrf.out
+echo nstasks=$SLURM_NTASKS
+date -u +%Y-%m-%d_%H:%M:%S >> wrf.times
+```
+
+## WRF correctness fix
```

Comment:
you had already mentioned the correctness patch above, the flow is not in sync with the code snippets

Reply:
I have moved this section of text up to before the build instructions.